src/gui/TrackContentObjectView.cpp

```diff
@@ -124,8 +124,13 @@ TrackContentObjectView::TrackContentObjectView( TrackContentObject * tco,
 			this, SLOT( updatePosition() ) );
 	connect( m_tco, SIGNAL( destroyedTCO() ), this, SLOT( close() ) );
 	setModel( m_tco );
-	connect( m_tco, SIGNAL( trackColorChanged() ), this, SLOT( update() ) );
-	connect( m_trackView->getTrackOperationsWidget(), SIGNAL( colorParented() ), this, SLOT( useTrackColor() ) );
+	connect(m_tco, SIGNAL(colorChanged()), this, SLOT(update()));
+
+	connect(m_trackView->getTrack(), &Track::colorChanged, this, [this]
+	{
+		// redraw if TCO uses track color
+		if (m_tco && !m_tco->usesCustomClipColor()) { update(); }
```

Comment:
Is it possible for `m_tco` to be null here? We don't test for it anywhere else in this file, and it doesn't really make sense to have a `TrackContentObjectView` that doesn't have a `TrackContentObject` to view.
```suggestion
		if (!m_tco->usesCustomClipColor()) { update(); }
```

Reply:
You're right, I forgot this after debugging what turned out to be a [three arg lambda](https://github.com/insilications/clazy-clr/blob/master/docs/checks/README-connect-3arg-lambda.md) segfault